docs/Inside_Arbitrum_Nitro/Inside_Arbitrum_Nitro.md

```diff
@@ -0,0 +1,527 @@
+
+### How the Sequencer Publishes the Sequence
+
+So how do nodes get the sequence? The sequencer publishes it in two ways: a real-time feed, and batches posted on L1 Ethereum.
+
+The real-time feed is published by the sequencer so that anyone who subscribes to the feed receives instant notification of each transaction as it is sequenced. Nitro nodes can subscribe to the feed directly from the sequencer, or through a relay that forwards the feed. The feed represents the sequencer's promise that it will record transactions in a particular order. If the sequencer is honest and doesn't have a long downtime, this promise will be kept. So anyone who trusts the sequencer to keep it promises can rely on the feed to get instant information about the transaction sequence--and they can run the sequenced transactions through the state transition function to learn the results of each transaction immediately. This is "soft finality" for transactions; it's "soft" because it depends on the sequencer keeping its promises.
+
+The sequencer also publishes its sequence on the L1 Ethereum chain. Periodically--perhaps every few minutes in production--the sequencer concatenates the next group of transactions in the feed, compresses them for efficiency, and posts the result as calldata on Ethereum. This is the final and official record of the transaction sequence. As soon as this Ethereum transaction has finality on Ethereum. the Layer 2 Nitro transactions it records will have finality. These transactions are final because their position in the sequence has finality, and the outcome of the transactions is deterministic and knowable to any party. This is "hard finality".
+
+The sequencer's batches are compressed using a general-purpose data compression algorithm called "brotli", on its highest-compression setting.
+
+## Geth at the Core
+
+The second key design idea in Nitro is "geth at the core." Here "geth" refers to go-ethereum, the most common node software for Ethereum. As its name would suggest, go-ethereum is written in the Go programming language, as is almost all of Nitro.
+
+![geth-sandwich](geth-sandwich.png)
+
+The software that makes up a Nitro node can be thought of as built in three main layers, which are shown above:
+
+* The base layer is the core of geth--the parts of geth that emulate the execution of EVM contracts and maintain the data structures that make up the Ethereum state. Nitro compiles in this code as a library, with a few minor modifications to add necessary hooks.
+* The middle layer, which we call ArbOS, is custom software that provides additional functions associated with Layer 2 functionality, such as decompressing and parsing the Sequencer's data batches, accounting for Layer 1 gas costs and collecting fees to reimburse for them, and supporting cross-chain bridge functionalities such as deposits of Ether and tokens from L1 and withdrawals of the same back to L1. We'll dig in to the details of ArbOS below.
```

Comment:
consistent capitalization for S/sequencer (I prefer Sequencer)

Reply:
Changed to Sequencer consistently.